### PR TITLE
Add dev.opengoal.OpenGOAL

### PR DESCRIPTION
**Pull request by Milkfat32**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.

Launcher for the OpenGOAL project, which brings back the GOAL language developed by Naughty Dog for the Playstation 2 series Jak and Daxter. This launcher makes it easier to use the OpenGOAL project and play the games with legitimate PS2 copies.

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*

Network is required in order to download the tooling from the OpenGOAL project repository for the games as the launcher and the OpenGOAL project have seperate releases. I have enabled both Wayland and X11 as this is the only combination that works without graphical glitches in the launcher and can launch the games. Using only X11 left trailing black shadows graphical glitch when moving the launcher window, but games still start correctly. With wayland and x11-fallback, only game sound is played, controller works but no game window is created. I can set x11 as the only one if required, but it was not the best experience of all options in my opinion.

- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: dev.opengoal.OpenGOAL.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        make-install-args:
+          - prefix=${FLATPAK_DEST}
+        sources:
+          - type: archive
+            url: https://dotat.at/prog/unifdef/unifdef-2.12.tar.xz
+            sha256: 43ce0f02ecdcdc723b2475575563ddb192e988c886d368260bc0a63aee3ac400
+        cleanup:
+          - '*'
+
+  - name: bzip2
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply dev.opengoal.OpenGOAL.yml:59]: also there is probably a cleanup rule missing to remove the headers / binaries / static archives.

File: dev.opengoal.OpenGOAL.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node20
+  - org.freedesktop.Sdk.Extension.rust-stable
+
+finish-args:
+  - --socket=wayland
+  - --socket=x11
+  - --device=all
+  - --socket=pulseaudio
+  - --share=network
```
Comment (hfiguiere) [inline dev.opengoal.OpenGOAL.yml:18]: `--share=ipc` is required to for X11

File: dev.opengoal.OpenGOAL.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Milkfat32) [reply dev.opengoal.OpenGOAL.yml]: Removed bzip2

File: dev.opengoal.OpenGOAL.yml (hunk 2 above)
Comment (Milkfat32) [reply dev.opengoal.OpenGOAL.yml:18]: Added

File: dev.opengoal.OpenGOAL.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>dev.opengoal.OpenGOAL</id>
+  
+  <name>OpenGOAL</name>
+  <summary>A launcher for the OpenGOAL Project to simplify usage and installation.</summary>
```
Comment (bbhtt) [inline dev.opengoal.OpenGOAL.appdata.xml:6]:
```suggestion
  <summary>Launcher for the OpenGOAL Project</summary>
```

This would pass the quality guidelines: must not begin with an article must not end with full stop.

File: dev.opengoal.OpenGOAL.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      Flatpak of the OpenGOAL Launcher. Legitimate copies of the games required in order to use.
+    </p>
+  </description>
+  
+  <launchable type="desktop-id">dev.opengoal.OpenGOAL.desktop</launchable>
+
+  <releases>
+      <release version="2.4.6" date="2024-06-24"/>
+  </releases>
+  <content_rating type="oars-1.0"/>
```
Comment (bbhtt) [inline dev.opengoal.OpenGOAL.appdata.xml:45]:
```suggestion
  <content_rating type="oars-1.1"/>
```

File: package-lock.json
Comment (bbhtt) [inline package-lock.json:1]: Why is this here?

File: yarn.lock
Comment (bbhtt) [inline yarn.lock:1]: Why is this here?

File: dev.opengoal.OpenGOAL.yml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      - type: git
+        url: https://github.com/open-goal/launcher.git
+        tag: v2.4.6
+        commit: a0294ac9e4459b0e2439e72fd4ea7364027e4def
+      - generated-sources.json
+      - cargo-sources.json
+      - type: file
+        path: dev.opengoal.OpenGOAL.appdata.xml
+        only-arches:
+          - x86_64
```
Comment (bbhtt) [inline dev.opengoal.OpenGOAL.yml:71]:
```suggestion
```

This source is not arch specific.

File: flathub.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (bbhtt) [inline flathub.json:2]: Why restrict to x86_64? You are building from source?

Comment (Milkfat32) [reply flathub.json:2]: Because the games currently only run on x86-64 which makes the launcher pointless for any other architecture

File: dev.opengoal.OpenGOAL.appdata.xml
Comment (bbhtt) [inline dev.opengoal.OpenGOAL.appdata.xml:1]: This should go upstream https://github.com/open-goal/launcher It's not specific to flatpak, you can include it with any packaging formats.

File: flathub.json (hunk 6 above)
Comment (bbhtt) [reply flathub.json:2]: Ok

File: dev.opengoal.OpenGOAL.appdata.xml
Comment (Milkfat32) [reply dev.opengoal.OpenGOAL.appdata.xml:1]: Of course, is it okay for the first release to have it in the flathub repo? Our ci did things to the file that did not happen during local testing, so I'll fix the ci for next release. I would rather not burden users with multiple auto updates in the launcher. If not, its understandable

Comment (bbhtt) [reply dev.opengoal.OpenGOAL.appdata.xml:1]: Ok fine.

**Comment by bbhtt** [comment]
bot, build dev.opengoal.OpenGOAL

File: dev.opengoal.OpenGOAL.yml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+command: open-goal-launcher
+
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node20
+  - org.freedesktop.Sdk.Extension.rust-stable
+
+finish-args:
+  - --socket=wayland
+  - --socket=x11
```
Comment (bbhtt) [inline dev.opengoal.OpenGOAL.yml:15]: Why is both x11 and wayland uses? If the application does not support wayland just use x11, otherwise use fallback-x11 instead of x11.

Comment (bbhtt) [reply dev.opengoal.OpenGOAL.yml:15]: Ah I missed the description:

> I have enabled both Wayland and X11 as this is the only combination that works without graphical glitches in the launcher and can launch the games. Using only X11 left trailing black shadows graphical glitch when moving the launcher window, but games still start correctly. With wayland and x11-fallback, only game sound is played, controller works but no game window is created. I can set x11 as the only one if required, but it was not the best experience of all options in my opinion.

Comment (bbhtt) [reply dev.opengoal.OpenGOAL.yml:15]: > Using only X11 left trailing black shadows graphical glitch when moving the launcher window, but games still start correctly. 

Was this being used on a Wayland desktop environment session? Ie, the window was using Xwayland? Or in Xorg.

Sounds like this doesn't quite work with Xwayland/Wayland

Comment (Milkfat32) [reply dev.opengoal.OpenGOAL.yml:15]: From my own testing this was what I thought was the best experience, as using only x11 the launcher worked, the game launch worked correctly, but anytime you moved the launcher window it left a trailing black shadow of the window until you stopped moving it. Using wayland and x11-fallback fixes the glitch but launching games did not work as you would only hear game sound no window was created for the game.

Having both wayland and x11 (no fallback) my system had no issues.

Comment (Milkfat32) [reply dev.opengoal.OpenGOAL.yml:15]: > > Using only X11 left trailing black shadows graphical glitch when moving the launcher window, but games still start correctly.
> 
> Was this being used on a Wayland desktop environment session? Ie, the window was using Xwayland? Or in Xorg session?
> 
> Sounds like this doesn't quite work with Xwayland/Wayland

Yes, the current game renderer is not quite up to par yet. Work is being done on this currently

Comment (bbhtt) [reply dev.opengoal.OpenGOAL.yml:15]: > both wayland and x11

The wayland permission is entirely unused unless you/the toolkit is forcing/auto switching the window to be native wayland eg. GDK_BACKEND/MOZ_ENABLE_WAYLAND etc.

X11 socket gives you xwayland on a Wayland session and x11 window on xorg.

So this sounds weird as to why it'd cause any difference at all. I can test once the build is posted.

**Comment by Milkfat32** [comment]
bot, build dev.opengoal.OpenGOAL

File: dev.opengoal.OpenGOAL.appdata.xml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+<component type="desktop-application">
+  <id>dev.opengoal.OpenGOAL</id>
+  <name>OpenGOAL</name>
+  <summary>Launcher for the OpenGOAL Project</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>ISC License</project_license>
+  <developer id="dev.opengoal">
+    <name>
+    OpenGOAL Contributors
+    </name>
```
Comment (bbhtt) [inline dev.opengoal.OpenGOAL.appdata.xml:11]:
```suggestion
    <name>
    OpenGOAL Contributors
    </name>
```
```suggestion
    <name>OpenGOAL Contributors</name>
```

**Comment by bbhtt** [comment]
Feel free to start the build if you need, you can validate the metainfo file with this command https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#validation

**Comment by Milkfat32** [comment]
bot, build dev.opengoal.OpenGOAL

File: dev.opengoal.OpenGOAL.appdata.xml [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  <url type="homepage">https://opengoal.dev/</url>
+  <url type="vcs-browser">https://github.com/open-goal/launcher</url>
+  <url type="faq">https://opengoal.dev/docs/faq</url>
+  <url type="translate">https://crowdin.com/project/opengoal-launcher</url>
+  <url type="contribute">https://opengoal.dev/docs/category/contributing</url>
+  <url type="contact">https://discord.gg/VZbXMHXzWv</url>
+  <screenshots>
+    <screenshot type="default">
+      <caption>Play the Jak and Daxter PlayStation 2 classics natively on your computer.</caption>
+      <image type="source">https://github.com/open-goal/open-goal.github.io/blob/master/documentation/usage/installation/img/launcher-install-a.png</image>
```
Comment (bbhtt) [inline dev.opengoal.OpenGOAL.appdata.xml:26]:
```suggestion
      <image type="source">https://raw.githubusercontent.com/open-goal/open-goal.github.io/master/documentation/usage/installation/img/launcher-install-a.png</image>
```

File: dev.opengoal.OpenGOAL.appdata.xml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    </screenshot>
+    <screenshot>
+      <caption>Need help? Just send a support package to the developers.</caption>
+      <image type="source">https://github.com/open-goal/open-goal.github.io/blob/master/documentation/usage/installation/img/launcher-install-a.png</image>
```
Comment (bbhtt) [inline dev.opengoal.OpenGOAL.appdata.xml:30]:
```suggestion
      <image type="source">https://raw.githubusercontent.com/open-goal/open-goal.github.io/b002bf0d317addc412635de7a537fbcf4a0c2252/documentation/usage/installation/img/launcher-install-a.png</image>
```

Comment (bbhtt) [reply dev.opengoal.OpenGOAL.appdata.xml:30]: This seems the same image as above?

File: dev.opengoal.OpenGOAL.appdata.xml (hunk 9 above)
Comment (bbhtt) [reply dev.opengoal.OpenGOAL.appdata.xml:26]: These should be direct links not links to html webpages.

File: dev.opengoal.OpenGOAL.appdata.xml (hunk 10 above)
Comment (Milkfat32) [reply dev.opengoal.OpenGOAL.appdata.xml]: Apologies, it went a little fast

**Comment by Milkfat32** [comment]
bot, build dev.opengoal.OpenGOAL

**Comment by bbhtt** [comment]
> Build [131150 successful](https://buildbot.flathub.org/#/builders/6/builds/131150) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/114151/dev.opengoal.OpenGOAL.flatpakref
> ```

Did you tests the build?

**Comment by Milkfat32** [comment]
> > Build [131150 successful](https://buildbot.flathub.org/#/builders/6/builds/131150) To test this build, install it from the testing repository:
> > ```
> > flatpak install --user https://dl.flathub.org/build-repo/114151/dev.opengoal.OpenGOAL.flatpakref
> > ```
> 
> Did you test the build?

Yes, it looks good to me.

**Comment by bbhtt** [comment]
Looks like I forgot to test the x11/wayland situation. Temporarily removing the ready label, I'll test it tomorrow.

File: dev.opengoal.OpenGOAL.yml (hunk 7 above)
Comment (bbhtt) [reply dev.opengoal.OpenGOAL.yml:15]: > as using only x11 the launcher worked, the game launch worked correctly, but anytime you moved the launcher window it left a trailing black shadow of the window until you stopped moving it.

I can't reproduce this. Seems to be working fine on GNOME 46/Fedora 40 wayland session using only `x11`


https://github.com/flathub/flathub/assets/62639087/6a51e8f4-439d-43b2-b0d8-817ed3818cfa

Comment (Milkfat32) [reply dev.opengoal.OpenGOAL.yml:15]: It is present on your recording just not as obvious on darker backgrounds.

Apologies for phone recording but I could not get it to show up on a screen recording. And like I said, it is minor but it is completely gone with wayland and x11 enabled. Other than that launcher and games work completely fine using only x11, just not wayland and x11 fallback.
[IMG_1535.webm](https://github.com/flathub/flathub/assets/26403463/a336a52f-aa4f-4b38-99e1-300acb598496)

Comment (bbhtt) [reply dev.opengoal.OpenGOAL.yml:15]: I think that was my VM being weird. Here it is on a lighter background in the VM.


https://github.com/flathub/flathub/assets/62639087/9de04f1f-6eb5-4a5a-8abc-33a8b5b3476b

Comment (Milkfat32) [reply dev.opengoal.OpenGOAL.yml:15]: Ok well sounds like an issue on my machine. I'll remove it from the manifest

**Comment by bbhtt** [comment]
bot, build dev.opengoal.OpenGOAL
